lib/pdksync.rb

```diff
@@ -332,6 +361,284 @@ def self.run_command(output_path, command)
     status.exitstatus
   end
 
+  # @summary
+  #   This method when called will find the source location of the gem to test
+  # @param [String] gem_to_test
+  #   The gem to test
+  # @param [String] gem_line
+  #   TThe line to update in the Gemfile
+  # @return [String]
+  #   The source location of the gem to test
+  def self.get_source_test_gem(gem_to_test, gem_line)
+    if !gem_line.nil?
+      new_data = gem_line.split(',')
+      return new_data # rubocop:disable Style/RedundantReturn
+    elsif !gem_to_test.nil?
+      file = File.open('Gemfile')
+      file.each_line do |line|
+        if line.include?(gem_to_test.to_s)
+          if line =~ %r{(http|https|ftp|ftps)\:\/\/[a-zA-Z0-9\-\.]+\.[a-zA-Z]{2,3}(\/\S*)?} # rubocop:disable Style/GuardClause
+            return line.split(',')[1].strip.to_s
+            break # rubocop:disable Lint/UnreachableCode
```

Comment:
The return line above negates the need for break  thus rubocop error.

Reply:
This is a better way without any nested if statements.

```ruby
def self.get_source_test_gem(gem_to_test, gem_line)
  return gem_line.split(",") if gem_line
  return gem_to_test unless gem_to_test
  gemfile_line = File.readlines("Gemfile").find do |line| 
    line.include?(gem_to_test.to_s) &&
  end
  return "https://github.com/puppetlabs/#{gem_to_test}" unless gemfile_line
  gemfile_line =~ %r{(http|https|ftp|ftps)\:\/\/[a-zA-Z0-9\-\.]+\.[a-zA-Z]{2,3}(\/\S*)?}
  line.split(",")[1].strip.to_s if line
end
```

You can also use the Bundler gem to perform all this work for you too.